modules/getNodePath.js

```diff
@@ -25,15 +27,44 @@ module.exports = function(type) {
 
     // global override?
     let globallySetType = Settings[`${type}Path`];
-    
+
     if (globallySetType) {
         resolvedPaths[type] = globallySetType;
     } else {
         var binPath = (Settings.inProductionMode)
             ? binaryPath.replace('nodes','node') + '/'+ type +'/'+ type
             : binaryPath + '/'+ type +'/'+ process.platform +'-'+ process.arch + '/'+ type;
 
-        if(Settings.inProductionMode) {
+        if(/*Settings.inProductionMode*/ true) {    
+            if(process.platform === 'linux' || 'darwin') {
+                var systemGethPath;
+                var proc = exec('which geth', 
+                    function(error, stdout, stderr) {
+                        systemGethPath = stdout.toString().replace(/\n/, '');
+                        if (error !== null) {
+                                log.error("error exec 'which geth': " + error + stdout);
+                        }
+                    });
+                
+                var systemGethVersion;
+                setTimeout(function () {
```

Comment:
I currently use `child_process.exec` to execute the two commands `which geth` and `geth version` and use a timeout to wait for the async call to finish. 
What better approaches are there? Especially considering the later program flow depends on its output?